pyigm/scripts/pyigm_mtlmcmc.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+"""
+This is the code that loop over data and runs mcmc chains.
+
+Written by Michele Fumagalli in Durham, Summer 2015
+michele.fumagalli@durham.ac.uk
+
+Copyright (C) 2015 Michele Fumagalli
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import pdb
+ 
+def run_mcmc(args):
+
+    from pyigm.metallicity.mcmc import mcmc_ions
+    import numpy as np
+    from astropy.table import Table
+
+    #load the file
+    sy = args.sy
+    data = Table.read(args.fileinput, format='ascii')
+
+    #find the ions for the current system
+    sets = np.where(data['name'] == sy.strip())
+    sets = sets[0]
+               
+    #now loop over ions for this system and initialise the tuples
+    observ=[]
+    for ii in range(len(sets)):
+        if(data['ion'][sets[ii]] == 'HI'):
+            #isolate hydrogen and info
+            #print ii, sets[ii], data['ion'][sets[ii]], sy
+            obsinfo={'NHI':data['logn'][sets[ii]],'eNHI':data['elogn'][sets[ii]],'hiflag':data['flag'][sets[ii]],
+                     'z':data['zabs'][sets[ii]],'errz':data['ezabs'][sets[ii]],'name':sy.strip()}
+        else:
+            #append ion value to list
+            observ.append((data['ion'][sets[ii]],data['logn'][sets[ii]],data['elogn'][sets[ii]],data['flag'][sets[ii]]))
+
+    #run the mcmc
+    print 'Ready to run mcmc for {}'.format(sy)
+
+    #pick optimised values for 12 processors - cosma (proc*NN walkers, proc*YY samples)
+    mcmc=mcmc_ions(observ,obsinfo,args.grid,nwalkers=(args.proc*80),nsamp=(args.proc*40),
```

Comment:
Ok, have put them in as optional.  I have also replaced number of
processors with nthread.